From d13bcf154d3791635e57c7a62c221a8e9eaf4993 Mon Sep 17 00:00:00 2001
From: Jens Reidel <adrian@travitia.xyz>
Date: Sun, 19 Jan 2025 23:12:19 +0100
Subject: [PATCH 16/17] Teach cc-rs about x86_64-chimera-linux-musl

Signed-off-by: Jens Reidel <adrian@travitia.xyz>
---
 vendor/cc-1.2.0/src/target/generated.rs | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/vendor/cc-1.2.0/src/target/generated.rs b/vendor/cc-1.2.0/src/target/generated.rs
index 0c9b0ae1a0..cff74861fa 100644
--- a/vendor/cc-1.2.0/src/target/generated.rs
+++ b/vendor/cc-1.2.0/src/target/generated.rs
@@ -2848,6 +2848,18 @@ pub(crate) const LIST: &[(&str, TargetInfo<'static>)] = &[
             unversioned_llvm_target: "x86_64-apple-watchos-simulator",
         },
     ),
+    (
+        "x86_64-chimera-linux-musl",
+        TargetInfo {
+            full_arch: "x86_64",
+            arch: "x86_64",
+            vendor: "unknown",
+            os: "linux",
+            env: "musl",
+            abi: "",
+            unversioned_llvm_target: "x86_64-unknown-linux-musl",
+        },
+    ),
     (
         "x86_64-fortanix-unknown-sgx",
         TargetInfo {
-- 
2.48.0

